src/mapdraw.c

```diff
@@ -1106,7 +1107,8 @@ int msDrawVectorLayer(mapObj *map, layerObj *layer, imageObj *image) {
 
   /* build item list. STYLEITEM javascript needs the shape attributes */
   if (layer->styleitem &&
-      (strncasecmp(layer->styleitem, "javascript://", 13) == 0)) {
+      ((STARTS_WITH_CI(layer->styleitem, "javascript://") == 0) ||
+       (STARTS_WITH_CI(layer->styleitem, "sld://") == 0))) {
```

Comment:
STARTS_WITH_CI() returns a boolean. You shouldn't compare to 0

```suggestion
      (STARTS_WITH_CI(layer->styleitem, "javascript://")  ||
       STARTS_WITH_CI(layer->styleitem, "sld://")) {
```